### PR TITLE
Implement Province Positions Loading

### PR DESCRIPTION
**Pull request by BrickPi**
Loads positions.txt information, tests passing.

### CODE REVIEW

**Comment by BrickPi** [comment]
@Spartan322 what's up with the Windows build?

**Comment by Spartan322** [comment]
You probably want to rebase this to master.

File: src/openvic-simulation/map/Province.cpp [hunk 1]
```diff
@@ -34,9 +34,62 @@ Province::life_rating_t Province::get_life_rating() const {
 }
 
 bool Province::load_positions(BuildingManager const& building_manager, ast::NodeCPtr root) {
-	// TODO - implement province position loading
-	// (root is the dictionary after the province identifier)
-	return true;
+	return expect_dictionary_keys(
+		"text_position", ZERO_OR_ONE, [this](ast::NodeCPtr node) -> bool {
+			return expect_fvec2(assign_variable_callback(positions.text))(node);
+		},
```
Comment (Hop311) [inline src/openvic-simulation/map/Province.cpp:40]:
```suggestion
		"text_position", ZERO_OR_ONE, expect_fvec2(assign_variable_callback(positions.text)),
```
The `expect` functions are already callbacks, no need to wrap them in a lambda function (this applies to every `expect` function in this PR).

File: src/openvic-simulation/map/Province.cpp [hunk 2]
```diff
@@ -34,9 +34,62 @@ Province::life_rating_t Province::get_life_rating() const {
...
+			"fort", ZERO_OR_ONE, [this](ast::NodeCPtr node) -> bool {
+				return expect_fixed_point(assign_variable_callback(positions.fort_rotation))(node);
+			},
+			"railroad", ZERO_OR_ONE, [this](ast::NodeCPtr node) -> bool {
+				return expect_fixed_point(assign_variable_callback(positions.railroad_rotation))(node);
+			},
+			"naval_base", ZERO_OR_ONE, [this](ast::NodeCPtr node) -> bool {
+				return expect_fixed_point(assign_variable_callback(positions.navalbase_rotation))(node);
+			},
+			"aeroplane_factory", ZERO_OR_ONE, [](ast::NodeCPtr _) -> bool { return true; } /* see below */
```
Comment (Hop311) [inline src/openvic-simulation/map/Province.cpp:86]:
```suggestion
			"aeroplane_factory", ZERO_OR_ONE, success_callback /* see below */
```
`success_callback`  is a `node_callback_t` that always returns `true`.

**Review by Hop311 — APPROVED** [review]

**Pull request merged**